lib/tests/test_lexer.c

```diff
@@ -155,8 +159,8 @@ _current_lloc(void)
 
 #define assert_parser_float(expected)                           \
   _next_token();                                                        \
-  assert_token_type(LL_FLOAT);                                         \
-  assert_true(_current_token()->fnum == expected, "Bad parsed value at %s:%d", __FUNCTION__, __LINE__);
+  cr_assert_token_type(LL_FLOAT);                                         \
+  cr_assert_float_eq(_current_token()->fnum, expected, 1e-12, "Bad parsed value at %s:%d", __FUNCTION__, __LINE__);
```

Comment:
Out of curiosity: how is this 1e-12 was chosen?

Reply:
I saw this value in one open source package. 

What do you suggest on epsilon value ?

Reply:
Yes you are right. Technically the floating point precision is more than 7, something like (24*log_2(10) = 7.225), so it's not worth go below 1e-7. From mathematical point of view, as this precision is just a precision of the representation, so any result of the arithmetic computation would have worse precision. Personally, I wouldn't have a problem even with a 1e-5, though in this specific assert does not involve arithmetics, just simple parsing, so 1e-7 is fine.